### PR TITLE
sources/azure: validate IMDS network configuration metadata

### PR DESCRIPTION
**Pull request by cjp256**
Due to race conditions and caching, IMDS may return stale or incomplete
metadata.  Add some validation to detect these scenarios and report
appropriate telemetry.

Signed-off-by: Chris Patterson <cpatterson@microsoft.com>

### CODE REVIEW

File: cloudinit/sources/DataSourceAzure.py [hunk 1]
```diff
@@ -178,6 +178,26 @@ def find_dev_from_busdev(camcontrol_out, busdev):
     return None
 
 
+def normalize_mac_address(mac: str):
+    """Normalize mac address with colons and lower-case."""
```
[the remark this answers is not in the document]
Comment (cjp256) [reply cloudinit/sources/DataSourceAzure.py:182]: I haven't added tests yet, this was just a draft for early review :D

I expect get_interfaces() will always be in this lower-case, colon format - but I figured I'd pass it through regardless.

IMDS metadata uses upper-case w/o colons, so it needs the normalization for comparison.

[the remark this answers is not in the document]
Comment (cjp256) [reply cloudinit/sources/DataSourceAzure.py:182]: We can use this method here too:
https://github.com/canonical/cloud-init/blob/c34f236b4675ca2a5136c8522417b4368f30b322/cloudinit/sources/DataSourceAzure.py#L2274

[the remark this answers is not in the document]
Comment (holmanb) [reply cloudinit/sources/DataSourceAzure.py:182]: Sounds good, thanks!

**Review by anhvoms — APPROVED** [review]
LGTM

**Review by TheRealFalcon — APPROVED** [review]
LGTM!

**Pull request merged**